modules/space/rendering/renderablefluxnodes.cpp

```diff
@@ -0,0 +1,1106 @@
+            ));
+            timeFile.clear();
+        }
+    }
+
+    if (timeFile.empty()) {
+        LERROR("Could not find a metadata file with time steps," 
+            " such as a csv, dat, txt or no file extention with 'time' in filename");
+    }
+
+    // time filestream
+    std::ifstream tfs(timeFile);
+    if (!tfs.is_open()) {
+        throw std::runtime_error("Could not open file");
+    }
+
+    std::string line;
+    std::getline(tfs, line);    //gets only first line
+    std::stringstream s;
+    s << line;
+
+    int nColumns = 0;
+    std::string columnName;
+    //loops through the names/columns in first line/header
+    while (s >> columnName) {
+        ++nColumns;
+    }
+    while (std::getline(tfs, line)) {   //for each line of data
+        std::istringstream iss(line);
+        for (int i = 0; i < nColumns; ++i) {    //for each column in line
+            std::string columnValue;
+            iss >> columnValue;
+            if (i != nColumns - 1) {    // last column
+                continue;
+            }
+            if (columnValue.length() == 23) {
+                // Ensure the separators are correct
+                columnValue.replace(4, 1, "-");
+                columnValue.replace(7, 1, "-");
+                columnValue.replace(13, 1, ":");
+                columnValue.replace(16, 1, ":");
+                columnValue.replace(19, 1, ".");
+                const double triggerTime = Time::convertTime(columnValue);
+                LDEBUG("timestring " + columnValue);
+                _startTimes.push_back(triggerTime);
+            }
+            else {
+                LERROR(fmt::format("Error in file formating. Last column in ",
+                    "file '{}' is not on UTC ISO8601 format", timeFile
+                ));
+            }
+            
+        }
+    }
+}
+
+void RenderableFluxNodes::updateActiveTriggerTimeIndex(double currentTime) {
+    auto iter = std::upper_bound(_startTimes.begin(), _startTimes.end(), currentTime);
+    if (iter != _startTimes.end()) {
+        if (iter != _startTimes.begin()) {
+            _activeTriggerTimeIndex = static_cast<int>(
+                std::distance(_startTimes.begin(), iter)
+            ) - 1;
+        }
+        else {
+            _activeTriggerTimeIndex = 0;
+        }
+    }
+    else {
+        _activeTriggerTimeIndex = static_cast<int>(_nStates) - 1;
+    }
+}
+void RenderableFluxNodes::render(const RenderData& data, RendererTasks&) {
+    if (_activeTriggerTimeIndex != -1) {
+        _shaderProgram->activate();
+
+        // Calculate Model View MatrixProjection
+        const glm::dmat4 rotMat = glm::dmat4(data.modelTransform.rotation);
+        const glm::dmat4 modelMat =
+            glm::translate(glm::dmat4(1.0), data.modelTransform.translation) *
+            rotMat *
+            glm::dmat4(glm::scale(
+                glm::dmat4(1.0), glm::dvec3(data.modelTransform.scale)
```

Comment:
The conversion is unnecessary;  `data.modelTransform.scale` is already a `glm::dvec3`